src/components/forms/Checkbox.jsx

```diff
@@ -0,0 +1,93 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import cx from 'classnames';
+
+import CheckEmptyIcon from '../../../lib/icons/react/CheckEmptyIcon';
+import CheckFilledIcon from '../../../lib/icons/react/CheckFilledIcon';
+import CheckIndeterminateIcon from '../../../lib/icons/react/CheckIndeterminateIcon';
+
+/**
+ * @param {Object} props react props
+ * @returns {ReactElement}
+ */
+const Checkbox = ({ name, label, value, onChange, indeterminate, checked, disabled }) => {
+  const [isChecked, setIsChecked] = useState(checked);
+
+  const handleChange = () => {
+    if (!disabled) {
+      const newState = !isChecked;
+      setIsChecked(newState);
+      onChange(newState);
```

Comment:
very very small nit: is there a better variable for `newState`? Like...  `updatedCheckedState` or `isCheckedUpdated` or something more contextual?